### PR TITLE
ESP32 - Update advertising data to be closer to the spec

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

This PR update the ble advertising data to get it closer to the spec.

 #### Summary of Changes
 * Update the advertising data on the esp32
 * Change the connection delegate to use the content of the advertising data instead of the connection name to connect to the peripheral
 * Change the code in examples/chip-tool/ to pass the discriminator instead of a name
 * Change the code in src/darwin to pass the discriminator instead of a name

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @jelderton @BroderickCarlin ?

**Review by saurabhst — APPROVED** [review]

File: src/platform/Darwin/BleConnectionDelegateImpl.mm [hunk 1]
```diff
@@ -115,12 +115,30 @@ - (void)centralManager:(CBCentralManager *)central
...
                      RSSI:(NSNumber *)RSSI
 {
-    BOOL isConnectable = !![advertisementData objectForKey:CBAdvertisementDataIsConnectable];
-    NSString * localNameKey = [advertisementData objectForKey:CBAdvertisementDataLocalNameKey];
-    if (isConnectable && [localNameKey isEqual:_deviceName]) {
-        ChipLogProgress(Ble, "Connecting to device: %@", _deviceName);
-        [self connect:peripheral];
-        [self stopScanning];
+    NSNumber * isConnectable = [advertisementData objectForKey:CBAdvertisementDataIsConnectable];
+    if ([isConnectable boolValue]) {
```
Comment (andy31415) [inline src/platform/Darwin/BleConnectionDelegateImpl.mm:119]: supernit: how do people feel about early return and continues to decrease code indents? I find it easier to understand code if it is less indented once it starts growing, however I also see a pattern in other chip places of 'only one exit point' (but does that apply to non-resource constrained code?)

like:

```
if (![isConnectable boolValue]) {
  return;  // only connectable devices are useful
}

CBUUID * .....
for (....) {
  if (!... isEqualToData...) { 
     continue;
  }

  if (....) { 
  }
}
```

vs.

```
if ([isConnectable boolValue]) {
  CBUUID * .....;

  for (....) {
     if (...) { 
       if (...) {
       }
    }
  }
}
```

**Review by jelderton — APPROVED** [review]

File: src/platform/ESP32/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -639,53 +647,25 @@ CHIP_ERROR BLEManagerImpl::ConfigureAdvertisingData(void)
...
+    advData[index++] = 0x06;                             // length
+    advData[index++] = CHIP_ADV_DATA_TYPE_SERVICE_DATA;  // AD type: (Service Data - 16-bit UUID)
+    advData[index++] = ShortUUID_CHIPoBLEService[0];     // AD value
+    advData[index++] = ShortUUID_CHIPoBLEService[1];     // AD value
+    advData[index++] = CHIP_BLE_OPCODE;                  // Used to differentiate from operational CHIP Ble adv
+    advData[index++] = advDataVersionDiscriminator >> 8; // Bits[15:12] == version - Bits[11:0] Discriminator
+    advData[index++] = advDataVersionDiscriminator & 0x00FF;
+
+    // Construct the Chip BLE Service Data to be sent in the scan response packet.
+    err = esp_ble_gap_config_adv_data_raw(advData, sizeof(advData));
```
Comment (andy31415) [inline src/platform/ESP32/BLEManagerImpl.cpp:665]: does the size always need to be sizeof == MAX_ADV_DATA_LEN or could we use index here to reduce payload sizes? does it matter?

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
Will merge since my comments were more nit/meh. Would love an answer, however if we need to adjust code we can do it so in a separate PR.

File: src/platform/Darwin/BleConnectionDelegateImpl.mm (hunk 1 above)
Comment (vivien-apple) [reply src/platform/Darwin/BleConnectionDelegateImpl.mm:119]: For my part I'm 100% in favour or early returns and to declared variables just before they are used, if possible.
I think @gerickson advocates for the 'single exit point' patterns thought.

File: src/platform/ESP32/BLEManagerImpl.cpp (hunk 2 above)
Comment (vivien-apple) [reply src/platform/ESP32/BLEManagerImpl.cpp:665]: At some point we will need to use manufacturer specific data and add them to the advertising data if there is enough space. The rest will go into the Scan Req/Resp packet.
I will dig into this when I will work on that in order to give you a more accurate answer.

**Pull request merged**